observer/lib/observer.js

```diff
@@ -37,3 +49,34 @@ export const observeTransferEvents = async (pgPool, ieContract, provider) => {
     await updateDailyTransferStats(pgPool, transferEvent, currentBlockNumber)
   }
 }
+
+/**
+ * Observe scheduled rewards on the Filecoin blockchain
+ * @param {import('pg').Pool} pgPool
+ * @param {import('ethers').Contract} ieContract
+ */
+const observeScheduledRewards = async (pgPool, ieContract) => {
+  console.log('Querying scheduled rewards from impact evaluator')
+  for (let i = 0; ; i++) {
+    let address
+    try {
+      address = await ieContract.readyForTransfer(i)
+    } catch (err) {
+      break
+    }
+    let scheduledRewards
+    try {
+      scheduledRewards = await ieContract.rewardsScheduledFor(address)
+    } catch (err) {
+      console.error('Error querying scheduled rewards for', address, { cause: err })
+      continue
+    }
+    console.log('Scheduled rewards for', address, scheduledRewards)
+    await pgPool.query(`
+      INSERT INTO daily_scheduled_rewards (day, address, scheduled_rewards)
+      VALUES (now(), $1, $2)
+      ON CONFLICT (day, address) DO UPDATE SET
+      scheduled_rewards = EXCLUDED.scheduled_rewards
+    `, [address, scheduledRewards])
+  }
```

Comment:
This is going to be very expensive. With 5k daily active participants (see [Spark Public Dashboard](https://spacemeridian.grafana.net/public-dashboards/32c03ae0d89748e3b08e0f08121caa14?orgId=1)), we are going to make 5k RPC API calls followed by 5k SQL queries.

(1)
Can we group RPC API calls and SQL queries into batches? I believe Ethers v6 supports request batching, and we are configuring the RPC API provider to use batching. Now, we need to find out how to trigger the batching behaviour.

What I would like to see at high level:

```js
const BATCH_SIZE = 10 // for example, we need to tweak this
for (const addrBatch of splitIntoBatches(dailyParticipantAddresses, BATCH_SIZE)) {
  // is this going to trigger Ethers.js batching behaviour?
  const rewards = await Promise.all(
    addrBatch.map(addr => await ieContract.rewardsScheduledFor(addr))
  )

  // run a single SQL query to update multiple rows
  await pgPool.query(`
    INSERT INTO daily_scheduled_rewards (day, address, scheduled_rewards)
    VALUES (now(), unnest($1::text[]), unnest($2::numeric[]))
    ON CONFLICT (day, address) DO UPDATE SET
    scheduled_rewards = EXCLUDED.scheduled_rewards
  `, [
    addrBatch,
    rewards
  ])
}
```

(The query is inspired by an existing query in spark-evaluate [here](https://github.com/filecoin-station/spark-evaluate/blob/bb459b2f2c5b57252a7de659e3827ec5655ec6de/lib/platform-stats.js#L32-L42)).

(2)
We must be mindful of how many requests we send to Glif RPC API. IMO, we shouldn't send all 5k queries as fast as the systems can handle as that would put too much load of the RPC API provider.

I propose to introduce a small delay between iterations of this loop.

If we batch requests for each 10 participants, we need to send ~500 requests. If we use a `1s` delay, then we will finish updating all participants in 500 seconds = 8.3 minutes. I think that's fast enough since the scheduled rewards are updated every 20 minutes.